python/tests/test_approximate_nearest_neighbors.py

```diff
@@ -625,10 +636,14 @@ def test_ivfpq(
 ) -> None:
     """
     (1) Currently the usePrecomputedTables is not used in cuml C++.
-    (2) ivfpq has become unstable in 24.10. It does not get passed with algoParam {"nlist" : 10, "nprobe" : 2, "M": 2, "n_bits": 4} in ci where test_ivfflat is run beforehand. avg_recall shows large variance, depending on the quantization accuracy. This can be fixed by increasing nlist, nprobe, M, and n_bits.
+
+    (2) ivfpq has become unstable in 24.10. It does not get passed with algoParam {"nlist" : 10, "nprobe" : 2, "M": 2, "n_bits": 4} in ci where test_ivfflat is run beforehand. avg_recall shows large variance, depending on the quantization accuracy. This can be fixed by increasing nlist, nprobe, M, and n_bits. Note ivf_pq is non-deterministic, and it seems due to kmeans initialization leveraging runtime values of GPU memory.
+
+    (3) If M is is too small (e.g. 2), the returned distances can be very different from the ground distances.
+    Spark rapids ml may give lower recall than cuvs sg because it aggregates local topk candidates by the returned distances.
```

Comment:
I thought cuml has a 'refine distances' api that can be used to fill in the exact distances.

Reply:
Yeah, that seems to be working for SG only. Do we want to support that or use that in our test case?

Reply:
But we invoke SG classes locally, so can we do the refine in each worker?  Seems important to accurately pick the top k from the partial results.

Reply:
Yeah, should be able to call the refine in mapInPandas and after cuvs PQ. A few questions:

(Q1) It seems the refine API returns the exact distances, together with the refined topk indices. The API does not return the approximate (PQ) distances to the refined topk indices. That means, our "ivf_pq" algorithm will be returning exact distances, which is different from the behavior of cuvs "ivf_pq". 

(Q2) Is the "ivf_pq + refine" equivalent to the ivf_pq build_algo of cagra? We have already supported the ivf_pq option of cagra.

Reply:
Added the refine API for ivf_pq to return exact distances.